src/vs/workbench/contrib/scm/common/scmService.ts

```diff
@@ -134,7 +134,7 @@ class SCMInput extends Disposable implements ISCMInput {
 		}
 
 		if (!transient) {
-			this.historyNavigator.add(this._value);
+			this.historyNavigator.replaceLast(this._value);
```

Comment:
FYI @meganrogge, I think this is what was intended because I ran into the same problem in chat